e2e-tests/src/test/finality_version.rs

```diff
@@ -0,0 +1,222 @@
+use aleph_client::{
+    pallets::{aleph::AlephSudoApi, elections::ElectionsApi, session::SessionApi},
+    utility::BlocksApi,
+    waiting::{AlephWaiting, BlockStatus},
+    AsConnection, TxStatus,
+};
+use anyhow::anyhow;
+use log::info;
+use primitives::{BlockNumber, SessionIndex, Version, LEGACY_FINALITY_VERSION};
+
+use crate::{
+    config::setup_test,
+    finality_version::{
+        check_finality_version_at_block, check_next_session_finality_version_at_block,
+    },
+};
+
+const UPGRADE_TO_VERSION: u32 = 1;
+const UPGRADE_SESSION: SessionIndex = 3;
+const UPGRADE_FINALIZATION_WAIT_SESSIONS: u32 = 3;
+
+const FIRST_INCOMING_FINALITY_VERSION: Version = 1;
```

Comment:
Shouldn't this be equal to `LEGACY_FINALITY_VERSION` rather than a hardcoded `1`? Although TBH I'm kinda confused what this test is exactly doing – I would expect the test to always try switching from `LEGACY_FINALITY_VERSION` to the next one, but... is that the case?

Reply:
Wait, shouldn't this be 2 even? Does this test actually change the version now?

Reply:
You are right - I missed this and it did not actually check the change. The first incoming version should be something different that `LEGACY_FINALITY_VERSION`. For the purpose of the test, it makes sense to set it to `LEGACY_FINALITY_VERSION + 1 = 2`.

Reply:
Should now be fixed.